### PR TITLE
web service to control robotic arm

### PR DESCRIPTION
**Pull request by minhthuc2502**
I added all APIs to control arm through web server.
See README

### CODE REVIEW

**Comment by GaelGuegan** [comment]
Please could you rename all your file, class, namespace, variables that you named **api or myapi** etc... by something more explicit like **rest or rest_http** or something better ?

**Comment by GaelGuegan** [comment]
Also it could be a good idea to follow a existing cpp coding style ? maybe @yo6snap have some preferences ? Do we need to send our c template to @minhthuc2502  ?

**Comment by minhthuc2502** [comment]
I just modified name of file, class, function in class more clearly.

**Comment by yo6snap** [comment]
Concerning C++ there is a nice coding style used by Google’s open-source projects. Details may be found here: https://google.github.io/styleguide/cppguide.html

**Comment by minhthuc2502** [comment]
I changed the way to name the file and function. I think that it is better and clearer

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
[the remark this answers is not in the document]
Comment (minhthuc2502) [reply .vscode/settings.json]: solved

File: includes/resthttp.hpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
+#ifndef __API_FOR_REST__
+#define __API_FOR_REST__
```
[the remark this answers is not in the document]
Comment (minhthuc2502) [reply includes/resthttp.hpp:2]: rename to  __REST_HTTP__

File: src/main.cpp [hunk 3]
```diff
@@ -59,43 +174,67 @@ int main(int argc, char *argv[]) {
...
 
-  tQueue.Close();
+    else if(!strcmp(mode,"server-web")){
+      TransmitionQueue tQueue;
 
-  LOG_I("Program end!");
+      LOG_I("Launching Server");
+      // for web service
+      struct MHD_Daemon * d;
+      d = MHD_start_daemon(MHD_USE_THREAD_PER_CONNECTION,
```
[the remark this answers is not in the document]
Comment (minhthuc2502) [reply src/main.cpp:227]: Yes,  HttpServer is a good name. I thought so as well but before, there is a problem that restricts me to do it. I will try again.

File: src/main.cpp [hunk 4]
```diff
@@ -59,43 +174,67 @@ int main(int argc, char *argv[]) {
...
-  }
+      tQueue.Close();
 
-  tQueue.Wait();
+      LOG_I("Program end!");
+    }
 
-  tQueue.Close();
+    else if(!strcmp(mode,"server-web")){
+      TransmitionQueue tQueue;
```
[the remark this answers is not in the document]
Comment (minhthuc2502) [reply src/main.cpp:222]: removed

File: src/main.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (minhthuc2502) [reply src/main.cpp]: I pushed the new commit. It solved the problem

**Comment by minhthuc2502** [comment]
I add the blank spaces in order to separate some module in function. It helps me see better. But if it is a coding style standard, i will avoid it.

**Comment by minhthuc2502** [comment]
removed blank spaces

**Comment by GaelGuegan** [comment]
Ah ok sorry I was thinking you let space but new line is okay.

**Pull request merged**